doc/src/special_topics/classification.rst

```diff
@@ -0,0 +1,100 @@
+===============================
+Classification of SymPy objects
+===============================
+
+There are several ways of how SymPy object is classified.
+
+class
+=====
+
+Like any other object in Python, SymPy expression is an instance of class. You can
+get the class of the object with built-in `type()` function, and check it with
+`isinstance()` function.
+
+    >>> from sympy import Add
+    >>> from sympy.abc import x,y
+    >>> type(x + y)
+    <class 'sympy.core.add.Add'>
+    >>> isinstance(x + y, Add)
+    True
+
+Classes represent only the programmatic structures of the objects, and does not
+distinguish the mathematical difference between them. For example, the integral
+of number and the integral of matrix both have the class `Integral`, although the
+former is number and the latter is matrix.
+
+    >>> from sympy import MatrixSymbol, Integral
+    >>> A = MatrixSymbol('A', 2, 2)
+    >>> type(Integral(1, x))
+    <class 'sympy.integrals.integrals.Integral'>
+    >>> type(Integral(A, x))
+    <class 'sympy.integrals.integrals.Integral'>
+
+kind
+====
+
+Kind indicates what mathematical object does the expression represent.
+You can retrieve the kind of expression with `kind()` function.
+
+    >>> from sympy import kind
+    >>> kind(Integral(1, x))
+    NumberKind
+    >>> kind(Integral(A, x))
+    MatrixKind(NumberKind)
+
+This result shows that `Integral(1, x)` is number, and `Integral(A, x)` is array (since matrix is 2-dimensional array) with number element.
```

Comment:
I guess this should say matrix rather than array.

Reply:
Oops